src/evo/deterministicmns.cpp

```diff
@@ -1026,8 +1026,14 @@ void CDeterministicMNManager::HandleQuorumCommitment(const llmq::CFinalCommitmen
 
 CDeterministicMNList CDeterministicMNManager::GetListForBlockInternal(gsl::not_null<const CBlockIndex*> pindex)
 {
-    AssertLockHeld(cs);
     CDeterministicMNList snapshot;
+
+    if (!DeploymentActiveAt(*pindex, Params().GetConsensus(), Consensus::DEPLOYMENT_DIP0003)) {
```

Comment:
should you add similar check also inside `while(true)` below?

```
pindex = pindex->pprev;
if (dip0003 is not activated -> no need to read snapshot)
```

Reply:
As soon as dip3 is active we enter that loop and write/cache the initial snapshot. You can never go below this snapshot in that loop so we can be pretty sure dip3 is always active there.